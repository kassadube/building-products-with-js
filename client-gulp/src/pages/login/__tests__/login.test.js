/* eslint-disable react/react-in-jsx-scope */

import configureMockStore from 'redux-mock-store';
import LoginWarraper, {Login} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: '2017-05-07T10:22:37.364Z',
};


test('# Login Warraper', () => {
  const store = mockStore({auth: user});
  const component = (<LoginWarraper user={user} store={store} />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});


test('# Login', () => {
  const component = (<Login />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});


test('# Login ', () => {
  const newLogin = 'test';
  const newPassword = '1234';  
  const token = 'asd';
  // const error = 'err';
  const navToHome = () => expect(true).toBeTruthy();
  const onLoginClick = ({login, password, rememberMe}) => {
    expect(login).toBe(newLogin);
    expect(password).toBe(newPassword);
    expect(rememberMe).toBeTruthy();
  };


  const component = (
    <Login user={user} onLoginClick={onLoginClick} token={token} navToHome={navToHome} />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
  const app = mount(component);
  // set new question text
  app.find('#usernameInput').getDOMNode().value = newLogin;
  // set new question expiration date
  app.find('#passwordInput').getDOMNode().value = newPassword;
  app.find('#rememberMe').getDOMNode().checked = true;
  // click answer button
  app.find('button').simulate('click');
});

