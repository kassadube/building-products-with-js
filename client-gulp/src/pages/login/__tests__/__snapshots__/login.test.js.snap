// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Login  1`] = `
<div
  className="jumbotron"
>
  <h3>
    Login Page
  </h3>
  <p>
    please log in Or 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/register"
    >
      register 
    </Link>
     
  </p>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="usernameInput"
      >
        Username
      </label>
      <input
        className="form-control"
        id="usernameInput"
        placeholder="username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="passwordInput"
      >
        Password
      </label>
      <input
        className="form-control"
        id="passwordInput"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="rememberMe"
      >
        <input
          id="rememberMe"
          name="rememberMe"
          type="checkbox"
        />
         Check me out
      </label>
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`# Login 1`] = `
<div
  className="jumbotron"
>
  <h3>
    Login Page
  </h3>
  <p>
    please log in Or 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/register"
    >
      register 
    </Link>
     
  </p>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="usernameInput"
      >
        Username
      </label>
      <input
        className="form-control"
        id="usernameInput"
        placeholder="username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="passwordInput"
      >
        Password
      </label>
      <input
        className="form-control"
        id="passwordInput"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="rememberMe"
      >
        <input
          id="rememberMe"
          name="rememberMe"
          type="checkbox"
        />
         Check me out
      </label>
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`# Login Warraper 1`] = `
<Login
  error=""
  navToHome={[Function]}
  onLoginClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  token=""
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": "2017-05-07T10:22:37.364Z",
    }
  }
/>
`;
